src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,93 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void add(T value) {
-
+        checkArraySize();
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkArraySize();
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Such index doesn't exist");
+        }
+        System.arraycopy(array, index, array, index + 1, size - 1);
```

Comment:
why `size - 1` as the last argument here?

Reply:
missclic)I looked at the wrong method.I needed do changed in remove()